src/Overlay.tsx

```diff
@@ -6,12 +7,30 @@ interface Props {
 }
 
 const Overlay = ({ data }: Props) => {
+
+  
+  /* Define local state hook to store the "user input" data */
+  const [playerData, setPlayerDataSwitch] = React.useState({player1: data.content.players[0],
+  player2: data.content.players[1]});
+
+  const switchPlayer = (e: { preventDefault: () => void; }) => {
+      /* Prevent button click's default behavior */
+      e.preventDefault();
+      debugger;
+      /* Call the state's "setter" method to update "userInput" state */
+      setPlayerDataSwitch(
+        {player1: data.content.players.find(p => p.team_index !== playerData.player1.team_index) || playerData.player1,
+         player2: data.content.players.find(p => p.team_index !== playerData.player2.team_index) || playerData.player2}
+         );
+  }
```

Comment:
if you need to use the previous state value to determine the new one use a callback like so :
```
setPlayerData((prev) => { player1: prev.player2, player2: prev.player1 })
```